src/store/index.ts

```diff
@@ -88,33 +102,70 @@ export default createStore<State>({
       if (res.status !== 200) {
         return "error";
       }
-      await dispatch("decryptUploadResponse", res.data);
+      const msg = await dispatch("decryptResponse", res.data);
```

Comment:
Afaik vuex results should not return values that is used. For this is probably makes more sense to just make `decryptResponse` a normal function, and call it from the relevant places, rather than an action, and pass in all the data it needs from the state as parameters.

Reply:
(Why shouldn't we return results from Actions?)

Should we move it to `cryptography.ts` instead? 
it makes sense to me to have it there.

Reply:
I am not 100% sure, but I don't think Vuex returns values (other than Promises that can be used to track completion) from actions. This might not be true in all cases.

Vuex follows a "one-way data flow" data flow model. From my understanding, once you start returning values from actions, you break the one-way flow model. So the goal is to separate actions on data from access to the data.